### PR TITLE
[ID-1118] Add provider configs to application.yml

### PR DESCRIPTION
**Pull request by samanehsan**
**Jira:** https://broadworkbench.atlassian.net/browse/ID-1118

**What:**
Move non-secret provider configuration values into `application.yml`

**Why:**
It is easier to view and update the provider configs in `application.yml` vs. in vault.

**How:**
- Add keys for each provider client_id and client_secret to vault
- Add remaining provider config values to `application.yml`.
- Update `./render_config`.sh to create a `secrets.env` file to export those values (instead of writing a `providers.yaml` file)

Note: This will need to be merged after [updating terra-helmfile](https://github.com/broadinstitute/terra-helmfile/pull/5226) to add environment-specific overrides. The only values that differ between environments at the moment are the `client_id`, `client_secret` and `allowedRedirectUriPatterns` for each provider.

### CODE REVIEW

**Review by Shakespeared — APPROVED** [review]

**Review by Ghost-in-a-Jar — APPROVED** [review]

File: service/src/main/resources/application.yml [hunk 1]
```diff
@@ -15,6 +15,31 @@ externalcreds:
...
+      externalIdClaim: "login"
+      issuer: "https://github.com"
+      linkLifespan: "15d"
+      scopes: ""
+      userNameAttributeName: "login"
+      revokeEndpoint: "https://api.github.com/installation/token"
+      userInfoEndpoint: "https://api.github.com/user"
+      authorizationEndpoint: "${externalcreds.providers.github.issuer}/login/oauth/authorize"
+      tokenEndpoint: "${externalcreds.providers.github.issuer}/login/oauth/access_token"
+      allowedRedirectUriPatterns: [ "https://bvdp-saturn-dev.appspot.com/oauth_callback" ]
```
Comment (Ghost-in-a-Jar) [inline service/src/main/resources/application.yml]: should the redirects be environment specific?

File: service/src/main/resources/application.yml [hunk 2]
```diff
@@ -15,6 +15,31 @@ externalcreds:
...
     key-ring-location: ${kMS_KEY_LOCATION:us-central1}
     ssh-key-pair-refresh-duration: ${KMS_KEY_REFRESH_DURATION:90d}
+  providers:
+    ras:
+      clientId: "${RAS_CLIENT_ID}"
+      clientSecret: "${RAS_CLIENT_SECRET}"
+      externalIdClaim: "preferred_username"
+      scopes: [ "openid","email","ga4gh_passport_v1","profile" ]
+      linkLifespan: "15d"
+      issuer: "https://stsstg.nih.gov"
```
Comment (Ghost-in-a-Jar) [inline service/src/main/resources/application.yml]: is there a non-prod env for ras?

File: render_config.sh [hunk 3]
```diff
@@ -8,13 +8,20 @@ COMMON_VAULT_PATH="secret/dsde/terra/kernel/$ENV/common"
...
   VAULT_COMMAND="docker run --rm -e VAULT_TOKEN=$VAULT_TOKEN -e VAULT_ADDR=$VAULT_ADDR vault:1.7.3 $VAULT_COMMAND"
 fi
 
-$VAULT_COMMAND -field=providers "$ECM_VAULT_PATH/providers" >"$SERVICE_OUTPUT_LOCATION/providers.yaml"
+{
+  echo export RAS_CLIENT_ID="$($VAULT_COMMAND -field=ras_client_id "$ECM_VAULT_PATH/providers")"
+  echo export RAS_CLIENT_SECRET="$($VAULT_COMMAND -field=ras_client_secret "$ECM_VAULT_PATH/providers")"
+  echo export GITHUB_CLIENT_ID="$($VAULT_COMMAND -field=github_client_id "$ECM_VAULT_PATH/providers")"
+  echo export GITHUB_CLIENT_SECRET="$($VAULT_COMMAND -field=github_client_secret "$ECM_VAULT_PATH/providers")"
+} >> "${SECRET_ENV_VARS_LOCATION}"
```
Comment (tlangs) [inline render_config.sh:23]: Since Fence secrets were added by @Ghost-in-a-Jar, should we include those here?

**Review by Ghost-in-a-Jar — APPROVED** [review]

File: render_config.sh (hunk 3 above)
Comment (samanehsan) [reply render_config.sh:23]: Oh you're right! I'll add those in

Comment (samanehsan) [reply render_config.sh:23]: Oh actually it looks like @Ghost-in-a-Jar started doing that in https://github.com/DataBiosphere/terra-external-credentials-manager/pull/179/files

**Review by tlangs — APPROVED** [review]

**Pull request merged**